test/JDBC/sql_expected/sys-server_principals.out

```diff
@@ -69,3 +69,34 @@ varchar#!#char#!#nvarchar#!#varchar#!#varchar
 jdbc_user#!#S#!#SQL_LOGIN#!#master#!#English
 ~~END~~
 
+
+select name, principal_id, sid, type, type_desc
+from sys.server_principals where name = 'public'
+GO
+~~START~~
+varchar#!#int#!#varbinary#!#char#!#nvarchar
+public#!#2#!#00000002#!#R#!#SERVER_ROLE
+~~END~~
+
+
+select name, principal_id, sid,
+suser_name(principal_id) as suser_name,
+suser_sid(name) as suser_sid,
+suser_sname(sid) as suser_sname,
+suser_sname(suser_sid(name)) as name2
+from sys.server_principals
+where name = 'public'
+GO
+~~START~~
+varchar#!#int#!#varbinary#!#nvarchar#!#varbinary#!#nvarchar#!#nvarchar
+public#!#2#!#00000002#!#public#!#00000002#!#public#!#public
+~~END~~
+
+
+select suser_sid('public')
```

Comment:
Add testcases where Public is passed with mixed cases(Upper + Lower case) if not already present.